R/simulationResultsToDataFrame.R

```diff
@@ -0,0 +1,66 @@
+#' Converts a `SimulationResults` objects to a data.frame
+#'
+#' @inheritParams getOutputValues
+#'
+#' @examples
+#' library(ospsuite)
+#'
+#' simPath <- system.file("extdata", "simple.pkml", package = "ospsuite")
+#' sim <- loadSimulation(simPath)
+#'
+#' # Running an individual simulation
+#' # results is an instance of `SimulationResults`
+#' results <- runSimulations(sim)
+#'
+#' # convert to a dataframe
+#' simulationResultsToDataFrame(results)
+#'
+#' @return
+#'
+#' SimulationResults object as data.frame with columns IndividualId, Time, paths,
+#' simulationValues, unit, dimension, TimeUnit, TimeDimension.
+#'
+#' @export
+
+simulationResultsToDataFrame <- function(simulationResults,
+                                         quantitiesOrPaths = NULL,
+                                         population = NULL,
+                                         individualIds = NULL) {
+  # no need to validating the input because this will be done by
+  # getOutputValues() anyways
+  simList <- getOutputValues(
+    simulationResults = simulationResults,
+    quantitiesOrPaths = quantitiesOrPaths,
+    population = population,
+    individualIds = individualIds
+  )
+
+  # convert data to long format with a new column for paths
```

Comment:
What does this means? What is a long format? 
This is probably obvious for you but this code  means nothing to me.  I think it would be beneficial for (me/us) to be tought a bit about what those funky methods are doing. I can't tell from the code what will happen (aside that IndividualId and Time column will be removed)

Reply:
Just to add my (unsolicited :) opinion: I think having a method that returns sim results in this long format is useful.  Even though the original results format is more concise, I've routinely had to convert to this long format when plotting and generating summaries.

Reply:
What is a long format? I am thinking of `LONG` as data type but this is probably not correct?

Reply:
Rather than have different columns for each path, have one `path` column, and specify the path for each row there.  Makes grouping for `ggplot` very quick and easy.